jquery-turtle.js

```diff
@@ -6444,6 +6444,50 @@ var turtlefn = {
     });
     return this;
   }),
+  copy: wrapcommand('copy', 0,
+  ["<u>copy()</u> makes a new turtle that is a copy of this turtle."],
+  function copy(cc) {
+    var t2 =  this.clone().insertAfter(this);
+    t2.hide();
+    this.plan(function(j, elem) { //t2.plan doesn't work here - 
+                                  //always applies to the most recent clone at the time of execution of plan!
+      cc.appear(j);
+      
+      //copy over turtle data:
+      olddata = getTurtleData(this);
+      newdata = getTurtleData(t2);
+      for (k in olddata) { newdata[k] = olddata[k]; }
+      
+      // copy over style attributes:
+      t2.attr('style', this.attr('style'));
+      
+      // copy each thing listed in css hooks:
+      for(property in $.cssHooks) {
+        var value = this.css(property);
+        t2.css(property, value);
+      }
+      
+      // copy attributes, just in case:
+      var attrs = this.prop("attributes");
+      //console.log(attrs)
+      for(i in attrs) {
+        t2.attr(attrs[i].name, attrs[i].value);
+      }
+      
+      // copy the canvas:
+      t2.canvas().width = this.canvas().width;
```

Comment:
David says: check if canvas returns anything before using it...

Also, consider adding a test with a non-canvas turtle.